### ISSUE TITLE
Question about Custom Loss Function Implementation

### ISSUE THREAD

**Issue by hira-afk** [IGlab-VUMC/MAGE_ab_generation#1]
I am currently studying a paper published in Cell that made significant contributions to antibody design.
"The loss function was modified to mask the antigen sequence so that the mean negative loglikelihood loss was calculated across the antibody sequence only during training (excluding the antigen sequence and [SEP])."

Code Observation
In the training script, I found a CustomNLLEntropyLossTrainer class with a masked_sep_loss function defined:

class CustomNLLEntropyLossTrainer(Trainer):
    　def __init__(self, *args, separator_token_id=31, **kwargs):
        　　super().__init__(*args, **kwargs)
     　   　self.separator_token_id = separator_token_id
    　def masked_sep_loss(model, batch, device):
       　 　targets = batch['input_ids']
      　  　# Custom loss implementation...

However, to the best of my understanding, this method appears to be defined but never called. The HuggingFace Trainer class uses the compute_loss method internally, which is not overridden in this implementation. I suspect that regular, unmasked training is being performed instead. Or could it be that the repository code and the paper do not fully correspond?

Thank you for your time and for making this code publicly available!
